tasks/main.yml

```diff
@@ -17,26 +12,237 @@
       {{ ansible_facts['distribution'] ~ '_' ~
       ansible_facts['distribution_version'] }}.yml
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __mssql_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __mssql_vars_file is file
+
+- name: "Verify that the user accepts EULA variables"
+  assert:
+    that:
+      - mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula
+      - mssql_accept_microsoft_cli_utilities_for_sql_server_eula
+      - mssql_accept_microsoft_sql_server_2019_standard_eula
+    fail_msg:
+      - "You must accept EULA by setting the following variables to true:"
+      - "mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula"
+      - "mssql_accept_microsoft_cli_utilities_for_sql_server_eula"
+      - "mssql_accept_microsoft_sql_server_2019_standard_eula"
+  register: __mssql_eula_evaluation
 
-# Examples of some tasks:
-- name: Ensure required packages are installed
+- name: Ensure that the tuned-profiles-mssql package is installed
   package:
-    name: "{{ __template_packages }}"
+    name: tuned-profiles-mssql
     state: present
 
-- name: Ensure required services are enabled and started
+- name: Ensure that the tuned service is started and enabled
   service:
-    name: "{{ item }}"
+    name: tuned
     state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
-
-- name: Generate /etc/{{ __template_foo_config }}
-  template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: yes
-    mode: 0400
-  notify: template handler to restart services
+    enabled: true
+
+- name: Get the active Tuned profile
+  command: tuned-adm active
+  changed_when: false
+  register: __mssql_tuned_active_profile
```

Comment:
This can return a list e.g. `mssql kernel-settings` https://github.com/linux-system-roles/kernel_settings/blob/master/library/kernel_settings.py#L170 and https://github.com/linux-system-roles/kernel_settings/blob/master/tasks/main.yml#L91

The role will have to do something like this:
```yaml

- name: Get the active Tuned profiles
  command: tuned-adm active
  changed_when: false
  register: __mssql_tuned_active_profiles

# adding it to the end of the list means that it will override any conflicting settings in other profiles
- name: Set the mssql Tuned profile
  command: tuned-adm profile '{{ __mssql_tuned_active_profiles.stdout }} mssql'
  when: '"mssql" not in __mssql_tuned_active_profiles.stdout'
```

Reply:
Thanks for pointing to this, the role indeed should not rewrite active profiles.
I implemented it but faced a bug that is caused by [BZ#1825882](https://bugzilla.redhat.com/show_bug.cgi?id=1825882): when setting profiles that have the same ancestor, the operation fails.
Examples:
```
[root@ibm-p8-kvm-03-guest-02 ~]# tuned-adm profile network-throughput mssql
Cannot load profile(s) 'network-throughput mssql': Cannot find profile 'throughput-performance' in '['/etc/tuned', '/usr/lib/tuned']'.
[root@ibm-p8-kvm-03-guest-02 ~]# tuned-adm profile throughput-performance mssql
Cannot load profile(s) 'throughput-performance mssql': Cannot find profile 'throughput-performance' in '['/etc/tuned', '/usr/lib/tuned']'.
[root@ibm-p8-kvm-03-guest-02 ~]# tuned-adm profile virtual-guest mssql
Cannot load profile(s) 'virtual-guest mssql': Cannot find profile 'throughput-performance' in '['/etc/tuned', '/usr/lib/tuned']'.
[root@ibm-p8-kvm-03-guest-02 ~]# tuned-adm profile virtual-host mssql
Cannot load profile(s) 'virtual-host mssql': Cannot find profile 'throughput-performance' in '['/etc/tuned', '/usr/lib/tuned']'.
```
I chatted with @yarda and he said that this bug is going to be fixed in RHEL 8.5 and is added to a Release Priority List.
I also find a way how to fix this locally and implemented this workaround. It will make the tests pass for now, and hopefully, I will be able to remove it when a fix is released to RHEL. How does this sound @richm?

Reply:
> How does this sound @richm?

sounds good